app/routes/endemics/vaccination.js

```diff
@@ -1,16 +1,56 @@
-const { urlPrefix } = require('../../config')
-const { endemicsVaccination } = require('../../config/routes')
+const Joi = require('joi')
+const session = require('../../session')
+const urlPrefix = require('../../config').urlPrefix
+const {
+  endemicsVaccination,
+  endemicsTestUrn,
+  endemicsVetRCVS,
+  endemicsTestResults
+} = require('../../config/routes')
+const { endemicsClaim: { herdVaccinationStatus: herdVaccinationStatusKey } } = require('../../session/keys')
+const radios = require('../models/form-component/radios')
 
 const pageUrl = `${urlPrefix}/${endemicsVaccination}`
 
-module.exports = [
-  {
-    method: 'GET',
-    path: pageUrl,
-    options: {
-      handler: async (request, h) => {
-        return h.view(endemicsVaccination)
+module.exports = [{
+  method: 'GET',
+  path: pageUrl,
+  options: {
+    handler: async (request, h) => {
+      const { latestVetVisitApplication, herdVaccinationStatus } = session.getEndemicsClaim(request)
+      const vaccinatedNotVaccinatedRadios = radios('', 'herdVaccinationStatus')([{ value: 'vaccinated', text: 'Vaccinated', checked: herdVaccinationStatus === 'vaccinated' }, { value: 'notVaccinated', text: 'Not vaccinated', checked: herdVaccinationStatus === 'notVaccinated' }])
+      const backLink = latestVetVisitApplication ? `${urlPrefix}/${endemicsVetRCVS}` : `${urlPrefix}/${endemicsTestResults}`
+      return h.view(endemicsVaccination, { backLink, ...vaccinatedNotVaccinatedRadios })
+    }
+  }
+}, {
+  method: 'POST',
+  path: pageUrl,
+  options: {
+    validate: {
+      payload: Joi.object({
+        herdVaccinationStatus: Joi.string().valid('vaccinated', 'notVaccinated').required()
+      }),
+      failAction: async (request, h, error) => {
+        const { latestVetVisitApplication } = session.getEndemicsClaim(request)
+        const vaccinatedNotVaccinatedRadios = radios('', 'herdVaccinationStatus', 'Select a vaccination status')([{ value: 'vaccinated', text: 'Vaccinated' }, { value: 'notVaccinated', text: 'Not vaccinated' }])
+        const backLink = latestVetVisitApplication === true ? `${urlPrefix}/${endemicsTestResults}` : `${urlPrefix}/${endemicsVetRCVS}`
```

Comment:
This only works if it is their first endemics claim and is based off the review from the old world. If they have done a new world review claim and then come through this for an endemics claim then this condition would still be true even though the old review is not relevant